packages/neuron-ui/src/widgets/Tabs/index.tsx

```diff
@@ -49,23 +53,41 @@ const DefaultVariant = <T extends any = {}>({
   )
 }
 
-export const Tabs = <T extends any = {}>({ tabs, Variant, onChange, ...rest }: TabsProps<T>) => {
+export const Tabs = <T extends {}, E extends {} = DefaultVariantProps>({
+  tabs,
+  Variant = DefaultVariant<T>,
+  onTabChange,
+  ...rest
+}: {
+  tabs: Tab<T>[]
+  Variant?: (props: VariantProps<T> & E) => ReactElement<any, any> | null
+  onTabChange?: (currentTab: Tab<T>) => void
+  // The finer-grained type of E is lost during destructuring, so a manual exclusion operation is performed to address this.
+} & Omit<E, keyof VariantProps<T>>) => {
   if (tabs.length === 0) {
     throw new Error('Tabs must have at least one tab')
   }
 
-  const [selectedTabId, setSelectedTabId] = useState<string | number>(tabs[0].id)
+  const [selectedTabId, setSelectedTabId] = useState<TabId>(tabs[0].id)
   const selectedTab = tabs.find(tab => tab.id === selectedTabId) ?? tabs[0]
```

Comment:
Yes

Reply:
`find` may potentially find an empty value because users can pass in a dynamic tabs array.
Therefore, using `tabs[0]` ensures that `selectedTab` is always non-empty.